dataikuapi/dss/apideployer.py

```diff
@@ -334,15 +334,29 @@ def start_update(self):
 
         return DSSFuture(self.client, future_response.get('jobId', None), future_response)
 
-    def delete(self):
+    def delete(self, disable_first=False):
         """
-        Deletes this deployment
+        Deletes this deployment. The disable_first flag automatically disables the deployment
+        before its deletion.
+
+        :param boolean disable_first: If True, automatically disables deployment.
+        If False, attempting to delete an enabled deployment will raise an Exception.
```

Comment:
Sorry for the extra nitpick, just realised that we tend to use `this deployment` in the rest of the doc, so just for consistency (+ a bit of a clearer message for the `False` case, I suggested something a bit too complicated I think)
```suggestion
        :param boolean disable_first: If True, automatically disables this deployment before deleting it.
        If False, will raise an Exception if this deployment is enabled
```